src/ApplicationConfig.php

```diff
@@ -23,50 +20,27 @@
     private $separator;
 
     /**
-     * @api
-     *
-     * @param array  $patterns
+     * @param array  $config
      * @param string $separator
-     *
-     * @return ApplicationConfig
      */
-    public static function fromFiles(array $patterns, $separator = '.')
+    public function __construct(array $config, $separator = '.')
     {
-        $locator = new FileLocator();
-        $files   = $locator->locate($patterns);
-
-        if (empty($files)) {
-            throw NoMatchingFilesException::fromPatterns($patterns);
-        }
-
-        $factory = new ReaderFactory([
-            '.json' => 'TomPHP\ConfigServiceProvider\FileReader\JSONFileReader',
-            '.php'  => 'TomPHP\ConfigServiceProvider\FileReader\PHPFileReader',
-        ]);
-
-        $configs = array_map(
-            function ($filename) use ($factory) {
-                $reader = $factory->create($filename);
-
-                return $reader->read($filename);
-            },
-            $files
-        );
-
-        $config = call_user_func_array('array_replace_recursive', $configs);
+        $this->config    = $config;
+        $this->separator = $separator;
```

Comment:
@tomphp 

Not sure, should we assert here that `$separator` is a single-line, trimmed, and not-empty `string`?